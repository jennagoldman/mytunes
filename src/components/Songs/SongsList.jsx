import React from 'react';
import { Link } from 'react-router-dom';
import SongItem from './SongItem.jsx';
import PropTypes from 'prop-types';

const SongsList = ({ title, songs, artist }) => {
  const songsListItems = songs.map(song => (
    <li key={song.id}>
      <Link to={`/song/${artist}/${song.title}`}>
        <SongItem {...song} />
      </Link>
    </li>
  ));

  return (
    <>
      <h2>{title} - Song List</h2>
      <ul>
        {songsListItems}
      </ul>
    </>
  );
};

SongsList.propTypes = {
  title: PropTypes.string.isRequired,
  songs: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default SongsList;
