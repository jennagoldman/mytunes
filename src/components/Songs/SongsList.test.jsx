import React from 'react';
import { shallow } from 'enzyme';
import SongsList from './SongsList.jsx';

describe('SongsList', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<SongsList 
      artist={'Electric Light Orchestra'}
      title={'Out of the Blue'}
      songs={[
        {
          title: 'Mr. Blue Sky',
          id: '789ghi'
        },
        {
          title: 'It\'s Over',
          id: '345jkl'
        }
      ]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
