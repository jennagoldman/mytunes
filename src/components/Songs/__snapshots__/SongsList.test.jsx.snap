// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongsList matches a snapshot 1`] = `
<Fragment>
  <h2>
    Out of the Blue
     - Song List
  </h2>
  <ul>
    <li
      key="789ghi"
    >
      <Link
        to="/song/Electric Light Orchestra/Mr. Blue Sky"
      >
        <SongItem
          id="789ghi"
          title="Mr. Blue Sky"
        />
      </Link>
    </li>
    <li
      key="345jkl"
    >
      <Link
        to="/song/Electric Light Orchestra/It's Over"
      >
        <SongItem
          id="345jkl"
          title="It's Over"
        />
      </Link>
    </li>
  </ul>
</Fragment>
`;
